app/code/Magento/CheckoutAgreements/Model/Agreement.php

```diff
@@ -9,6 +9,16 @@
 
 class Agreement extends \Magento\Framework\Model\AbstractExtensibleModel implements AgreementInterface
 {
+    /**
+     * @var string
+     */
+    protected $_eventPrefix = 'sales_agreement';
```

Comment:
I think we should name it "checkout_agreement"

Reply:
Updated as requested